src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -33,14 +35,17 @@ def __init__(self, env_layer, execution_config, composite_logger, telemetry_writ
         self.cmd_clean_cache = "sudo tdnf clean expire-cache"
         self.cmd_repo_refresh = "sudo tdnf -q list updates"
 
+        # fetch snapshottime from health_store_id
+        self.snapshot_posix_time = self.__get_posix_time(execution_config.max_patch_publish_date, env_layer)
+
         # Support to get updates and their dependencies
-        self.tdnf_check = 'sudo tdnf -q list updates'
-        self.single_package_check_versions = 'sudo tdnf list available <PACKAGE-NAME>'
-        self.single_package_check_installed = 'sudo tdnf list installed <PACKAGE-NAME>'
-        self.single_package_upgrade_simulation_cmd = 'sudo tdnf install --assumeno --skip-broken '
+        self.tdnf_check = self.__generate_command_with_snapshottime('sudo tdnf -q list updates <SNAPSHOTTIME>', self.snapshot_posix_time)
+        self.single_package_check_versions = self.__generate_command_with_snapshottime('sudo tdnf list available <PACKAGE-NAME> <SNAPSHOTTIME>', self.snapshot_posix_time)
+        self.single_package_check_installed = self.__generate_command_with_snapshottime('sudo tdnf list installed <PACKAGE-NAME> <SNAPSHOTTIME>', self.snapshot_posix_time)
+        self.single_package_upgrade_simulation_cmd = self.__generate_command_with_snapshottime('sudo tdnf install --assumeno --skip-broken <SNAPSHOTTIME>', self.snapshot_posix_time)
 
         # Install update
-        self.single_package_upgrade_cmd = 'sudo tdnf -y install --skip-broken '
+        self.single_package_upgrade_cmd = self.__generate_command_with_snapshottime('sudo tdnf -y install --skip-broken <SNAPSHOTTIME>', self.snapshot_posix_time)
```

Comment:
self.__ ? Are these expected to be accessible outside the class?

Reply:
Assessment is not supposed to be affected by being part of an SDP install patches flow.
It'll just yo-yo the assessment information between that operation and the next CP, AP or non-SDP IP operation.
Please review the Apt implementation a bit closer for the strict SDP parallels.

Reply:
@kjohn-msft I think snapshottime should also include snapshottime, because the list of patches available changes based on snapshottime and this is majorly be relevant in later SDP stages. For eg: A rollout starting 5/23, will have that snapshottime affixed. Now, say, this will reach say phase 10 after 8 days, there may be more patches available to install after 8 days than on 5/23. In this case, if we don't use snapshottime in all assessment queries, we end up listing more patches as available to the customer than what we install. This is raise CRIs on not all patches are being installed.